model/env.go

```diff
@@ -0,0 +1,51 @@
+package model
+
+import (
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// EnvVar contains the value source for a given environement variable.
+type EnvVar struct {
+	Value     string               `json:"value,omitempty"`
+	ValueFrom *corev1.EnvVarSource `json:"valueFrom,omitempty"`
+}
+
+// EnvVarMap is a map of multiple env var names to their values.
+type EnvVarMap map[string]EnvVar
```

Comment:
0/5 should EnvVarMap be a `map[string]*EnvVar` instead?

Reply:
Either seems fine, but the only reason I am 0.5/5 on this is because this method made it cleaner to use elsewhere in the code. I will change it if you think it's better that way.

Reply:
No, whatever makes it cleaner. I guess in theory if this map got big there might be performance implications but that's probably a premature optimization on my part.